moldesign/models/lammps.py

```diff
@@ -0,0 +1,370 @@
+# Copyright 2016 Autodesk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cStringIO import StringIO
+
+import moldesign.molecules
+from moldesign import compute
+from moldesign import forcefields as ff
+
+from moldesign.utils import from_filepath
+
+from lammps import lammps, PyLammps
+
+def exports(o):
+    __all__.append(o.__name__)
+    return o
+__all__ = []
+
+
+@exports
+class LAMMPSPotential(EnergyModelBase):
+    """Creates an OpenMM "context" to drive energy calculations.
+    Note that, while a dummy integrator is assigned, a different context will
+    be created for any MD calculations.
+
+    :ivar sim: openmm simulation object
+    :type sim: simtk.openmm.app.Simulation
+    """
+    # NEWFEATURE: need to set/get platform (and properties, e.g. number of threads)
+    DEFAULT_PROPERTIES = ['potential_energy', 'forces']
+    
+    PARAMETERS = [
+        mdt.parameters.Parameter(
+            'skin', 'Extra distance beyond force cutoff',
+            type=u.angstrom, default=2.0)]
+
+    
+    # calculate potential energy and force
+    def calculate(self, requests):
+        
+        lammps_system = self.lammps_system
+        lammps_system.run(500)
```

Comment:
This part should be in the integrator - we won't do any running in the EnergyModel